tests/Unit/classes/ValidateCoreTest.php

```diff
@@ -169,6 +177,21 @@ public function isBirthDateProvider()
         );
     }
 
+    public function isDateOrNullProvider()
+    {
+        return array(
+            array(true, '1991-04-19'),
+            array(true, '2015-03-22'),
+            array(true, '1945-07-25'),
+            array(true, '2020-03-19'),
+            array(false, '1991-03-33'),
+            array(false, '1991-15-19'),
+            array(true, null),
```

Comment:
Shouldn't it also test dates with time '2015-03-22 32:00:00' ?

Reply:
You're right. I just added some tests with the time.